base_layer/key_manager/src/birthday.rs

```diff
@@ -0,0 +1,140 @@
+// Copyright 2021. The Tari Project
+//
+// Redistribution and use in source and binary forms, with or without modification, are permitted provided that the
+// following conditions are met:
+//
+// 1. Redistributions of source code must retain the above copyright notice, this list of conditions and the following
+// disclaimer.
+//
+// 2. Redistributions in binary form must reproduce the above copyright notice, this list of conditions and the
+// following disclaimer in the documentation and/or other materials provided with the distribution.
+//
+// 3. Neither the name of the copyright holder nor the names of its contributors may be used to endorse or promote
+// products derived from this software without specific prior written permission.
+//
+// THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES,
+// INCLUDING, BUT NOT LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+// DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL,
+// SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR
+// SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY,
+// WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE
+// USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+
+use std::convert::TryFrom;
+
+use serde::{Deserialize, Serialize};
+use tari_common::configuration::Network;
+use tari_core::blocks::genesis_block::get_genesis_block;
+
+#[derive(Clone, Debug, PartialEq, Eq, Serialize, Deserialize)]
+pub struct Birthday {
+    zero_point_time: u64,
+    birthday: u16,
+    version: u8,
+}
+
+impl Birthday {
+    pub fn default() -> Self {
+        let network = Network::Dibbler;
+        Self::new(network)
+    }
+
+    pub fn new(network: Network) -> Self {
+        let current_time = Self::current_time_in_seconds();
+        Self::new_from_current_time(network, current_time)
+    }
+
+    fn new_from_current_time(network: Network, current_time: u64) -> Self {
```

Comment:
```suggestion
    fn new_from_time(network: Network, time: u64) -> Self {
```

if it was the current time you were using, you wouldn't need to pass the time in

Reply:
I totally agree. I only implemented this way for testing purposes. I am going to refactor it conveniently.